File: dfs/programmers/level3/43162. 네트워크/43162.py
def solution(n, computers):
    def dfs(i):
        visited[i] = True
        for j in range(n):
            if visited[j] == False and computers[i][j] == 1:
                dfs(j)

    answer = 0
    
    visited = [False] * n
    
    for i in range(n):
        if visited[i] == False:
            dfs(i)
            answer += 1
        
    
    return answer

File: dfs/programmers/level3/43162. 네트워크/43162.java
class Solution {
    private boolean[] visited;
    private int[][] computer;
    
    private void dfs(int node) {
        visited[node] = true;  
         for (int i = 0; i < computer.length; i++) {
            if (computer[node][i] == 1 && !visited[i]) { // 연결된 컴퓨터가 있고, 아직 방문하지 않았다면
                dfs(i);
            }
        }
        
    }
    
    public int solution(int n, int[][] computers) {
        computer = computers;
        visited = new boolean[n];
        int count = 0;
        
        for(int i = 0; i < n; i++) {
            if (!visited[i]) {  // 방문하지 않은 컴퓨터라면 DFS 수행
                dfs(i);
                count++; 
            }
        }
        
        return count;
    }
}
